modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -161,6 +161,26 @@ sealed abstract class Json extends Product with Serializable {
    * Use implementations provided by case classes.
    */
   override def hashCode(): Int
+
+
+  /**
+    * Recursively return all values matching the specified `key`.
+    *
+    * The Play docs, from which this method was inspired, reads:
+    *   "Lookup for fieldName in the current object and all descendants."
+    */
+  final def \\(key: String): List[Json] = keyValues(this).collect {
+    case (k, v) if (k == key) => v
+  }
+
+private def keyValues(json: Json): List[(String, Json)] = json match {
+    case JObject(obj)  => obj.toList.flatMap { case (k, v) => keyValuesHelper(k, v) }
+    case JArray(elems) => elems.toList.flatMap(keyValues)
+    case _             => List.empty
```

Comment:
This is just a matter of nitpicky consistency with the rest of the codebase, but could we use `Nil` here.